### PR TITLE
[clarity] Add script for flow unit tests

### LINKED ISSUE #519: [BLOCKER] Clarity Native Testing Better Annotations

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a script to handle flow tests in clarity
* adds a simple test for signer-pre-register and signer-register function calls in sbtc-stacking-pool
* fixes #519 

Flow tests are tests in filesname suffix `_flow_tests`. They
* can not contain nested parenthesis, only `(begin` followed by a sequence of function calls.
* can contain function calls that must proceed with at least on annotation
* can only contain 
  * calls to a public function within the contract without parameters or  
  * contract calls using `(contract-call?` wrapped in `unwrap!` 
* function call annotations can be `mine-blocks-before`, `new-block`

Clarity contract calls are converted into `chain.mineBlock` expressions.

### CODE REVIEW

**Comment by friedger** [comment]
Improves code coverage already from 12% to 35% of sbt-stacking-pool

**Comment by jo-tm** [comment]
please review @MarvinJanssen @setzeus @jo-tm @FriendsFerdinand

File: sbtc-mini/ext/generate-flow-tests.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+      Deno.writeTextFile(
+        `${targetFolder}/${contractName}.ts`,
+        code.flat().join("\n")
+      );
+    }
+  },
+});
+
+function generateTxString(
```
Comment (jo-tm) [inline sbtc-mini/ext/generate-flow-tests.ts:138]: can we document this  a little more?

File: sbtc-mini/ext/generate-flow-tests.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    })
+    .join(", ");
+
+  return `Tx.contractCall("${callInfo.contractName || contractPrincipal}", "${
+    callInfo.functionName
+  }", [${argStrings}], callerAddress)
+  `;
+}
+
+function generateBlocks(contractPrincipal: string, calls: FunctionBody) {
```
Comment (jo-tm) [inline sbtc-mini/ext/generate-flow-tests.ts:179]: more documentation please.

**Review by jo-tm — APPROVED** [review]
approved with reservations on docs, needs more doc

**Review by setzeus — APPROVED** [review]
Pulled & confirmed all 75 unit tests & 1 flow test passing.

File: sbtc-mini/ext/generate-flow-tests.ts (hunk 1 above)
Comment (friedger) [reply sbtc-mini/ext/generate-flow-tests.ts]: done in [85c366f](https://github.com/Trust-Machines/stacks-sbtc/pull/613/commits/85c366fc504e92cb640ce8d9f09e26e70bf61df2)

File: sbtc-mini/ext/generate-flow-tests.ts (hunk 2 above)
Comment (friedger) [reply sbtc-mini/ext/generate-flow-tests.ts:138]: done in [85c366f](https://github.com/Trust-Machines/stacks-sbtc/pull/613/commits/85c366fc504e92cb640ce8d9f09e26e70bf61df2)

**Pull request merged**